provider/execution/execution.go

```diff
@@ -107,11 +107,14 @@ func (e TableExecutor) doMultiplexResolve(ctx context.Context, clients []schema.
 	logger := clients[0].Logger()
 	logger.Debug("multiplexing client", "count", len(clients), "table", e.Table.Name)
 	defer close(diagsChan)
+	numberOfClients := 0
```

Comment:
I initially used a wait group, maybe its smarter, but on the other hand adding another sync primitive is unwise.